diagnosis/getkuberneteslogs.sh

```diff
@@ -23,6 +23,151 @@ function restore_ssh_config
     fi
 }
 
+retrycmd_if_failure() 
+{ 
+    retries=$1; 
+    wait=$2; 
+    for i in $(seq 1 $retries); do 
+        ${@:3}; [ $? -eq 0  ] && break || sleep $wait; 
+    done; 
+    log_level -i "Command Executed $i times."; 
+}
+
+
+function login_azs()
+{
+    local spn_id=$1
+    local spn_secret=$2
+    local tenant_id=$3
+    
+    az login --service-principal -u $spn_id -p $spn_secret --tenant $tenant_id
+}
+
+askSubscription() {
+    az account list -otable
+    echo ""
+    echo "Please enter the Id of the account you wish to use. If you do not see"
+    echo "a valid account in the list press Ctrl+C to abort and create one."
+    echo "If you leave this blank we will use the Current account."
+    echo -n "> "
+    read azure_subscription_id
+
+    if [ "$azure_subscription_id" != "" ]; then
+        az account set --subscription $azure_subscription_id
+    else
+        azure_subscription_id=$(az account list --output json | jq -r '.[] | select(.isDefault==true) | .id')
+    fi
+    echo "Using subscription_id: $azure_subscription_id"
+}
+
+askName() {
+    echo ""
+    echo "Choose a name for your resource group, storage account and client"
+    echo "client. This is arbitrary, but it must not already be in use by"
+    echo "any of those resources. ALPHANUMERIC ONLY. Ex: mypackerbuild"
+    echo -n "> "
+    read meta_name
+}
+
+createResourceGroup() {
+    local location=$1
+    echo "==> Creating resource group"
+    az group create -n $meta_name -l $location
+    if [ $? -eq 0 ]; then
+        azure_group_name=$meta_name
+    else
+        echo "Error creating resource group: $meta_name"
+        return 1
+    fi
+}
+
+createStorageAccount() {
+
+    local location=$1 
+
+    echo "==> Creating storage account"
+    az storage account create --name $meta_name --resource-group $meta_name --location $location --kind Storage --sku Standard_LRS
+    if [ $? -eq 0 ]; then
+        azure_storage_name=$meta_name
+    else
+        echo "Error creating storage account: $meta_name"
+        return 1
+    fi
+}
+
+
+retryable() {
+    n=0
+    until [ $n -ge $1 ]
+    do
+        $2 && return 0
+        echo "$2 failed. Retrying..."
+        n=$[$n+1]
+        doSleep
+    done
+    echo "$2 failed after $1 tries. Exiting."
+    exit 1
+}
+
+requirements() {
+    found=0
+
+    azureversion=$(az --version)
+    if [ $? -eq 0 ]; then
+        found=$((found + 1))
+        echo "Found azure-cli version: $azureversion"
+    else
+        echo "azure-cli is missing. Please install azure-cli from"
+        echo "https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest"
```

Comment:
the link that explains how to set up azure-cli for azure stack maybe a better option